sphinx/environment/__init__.py

```diff
@@ -692,6 +691,7 @@ def read_doc(self, docname, app=None):
                 break
         else:
             self.settings['smart_quotes'] = False
+        self.settings['smartquotes_action'] = self.config.smartquotes_action
```

Comment:
Docutils has no smartquotes_action user config setting yet; but it introduced the docutils.transforms.universal.SmartQuotes class attribute at 0.14, and it is useful for us, because Sphinx users may want to for example not apply smart quotes to ellipsis conversion (from  ``...``). For Docutils < 0.14, there was at some point some bad config which educated backticks also, and we override that in our own sphinx.util.smartypants (this module is used by Sphinx only for Docutils < 0.14).

Reply:
Is there any reason for setting this into `self.settings`?

Reply:
It was in anticipation that Docutils might offer such an option in future to configure the `smartquotes_action` attribute of its docutils.transforms.universal.SmartQuotes, which we inherit for SphinxSmartQuotes. Also, I did this before RSTParser inherited from Parser hence got access to app. Thinking twice, it might indeed be better not to set this into ``self.settings``, else we will again have problem that a Docutils config file would override the Sphinx user conf.py, if such an option is introduced in future by Docutils.

Reply:
If docutils does not implement the option yet, it might be too premature action. -1 for adding key.

Reply:
I have removed it from ``self.settings``, but I do keep it as confval because it is useful for Sphinx users. Is that ok?